index.js

```diff
@@ -1053,9 +1066,10 @@
                       constraints,
                       expArgTypes.concat([expRetType]),
                       values[idx],
-                      index,
+                      isNaN(index) ? expArgTypes.length : index,
```

Comment:
How does this look, Kevin? It seems preferable to handle `NaN` sooner rather than later.